include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,194 @@
+  common::RoundRobin<matrix::Panel<Coord::Col, T, Device::CPU>> panelsW(n_workspaces,
+                                                                        mat_v.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Row, T, Device::CPU>> panelsW2(n_workspaces,
+                                                                         mat_c.distribution());
+
+  dlaf::matrix::Distribution dist_t({mb, total_reflector}, {mb, mb});
+  matrix::Panel<Coord::Row, T, Device::CPU> panelT(dist_t);
+
+  const SizeType total_reflector_blocks = dist_t.nrTiles().cols();
+  const SizeType total_reflectors_last_block =
+      dist_t.tileSize(GlobalTileIndex(0, total_reflector_blocks - 1)).cols();
+
+  for (SizeType k = total_reflector_blocks - 1; k >= 0; --k) {
+    bool is_last = (k == total_reflector_blocks - 1);
+    const GlobalTileIndex v_start{k + 1, k};
+    const LocalTileIndex kk{k, k};
+
+    auto& panelV = panelsV.nextResource();
+    auto& panelW = panelsW.nextResource();
+    auto& panelW2 = panelsW2.nextResource();
+
+    panelV.setRangeStart(v_start);
+    panelW.setRangeStart(v_start);
+
+    if (is_last) {
+      panelT.setHeight(total_reflectors_last_block);
+      panelW2.setHeight(total_reflectors_last_block);
+      panelW.setWidth(total_reflectors_last_block);
+      panelV.setWidth(total_reflectors_last_block);
+    }
+
+    for (SizeType i = k + 1; i < mat_v.nrTiles().rows(); ++i) {
+      auto ik = LocalTileIndex{i, k};
+      if (i == k + 1) {
+        hpx::shared_future<matrix::Tile<const T, Device::CPU>> tile_v = mat_v.read(ik);
+        if (is_last) {
+          tile_v = splitTile(tile_v, {{0, 0},
+                                      {mat_v.distribution().tileSize(GlobalTileIndex(i, k)).rows(),
+                                       total_reflectors_last_block}});
+        }
+        copySingleTile(tile_v, panelV(ik));
+        hpx::dataflow(hpx::launch::sync, unwrapping(tile::laset<T>), lapack::MatrixType::Upper, 0.f, 1.f,
+                      panelV(ik));
+      }
+      else {
+        panelV.setTile(ik, mat_v.read(ik));
+      }
+    }
+
+    auto taus_panel = taus[k];
+    const SizeType nr_reflectors = (is_last) ? total_reflectors_last_block : mat_v.blockSize().cols();
+    const LocalTileIndex k_factor{Coord::Col, k};
+
+    dlaf::factorization::internal::computeTFactor<Backend::MC>(nr_reflectors, mat_v, v_start, taus_panel,
+                                                               panelT(k_factor));
+
+    // W = V T
+    hpx::shared_future<matrix::Tile<const T, Device::CPU>> tile_t = panelT.read(k_factor);
+    for (const auto& idx : panelW.iteratorLocal()) {
+      copySingleTile(panelV.read(idx), panelW(idx));
+      trmmPanel(executor_np, tile_t, panelW(idx));
+    }
+
+    // W2 = W C
+    matrix::util::set0(executor_hp, panelW2);
+    LocalTileIndex c_start{k + 1, 0};
+    LocalTileIndex c_end{m, n};
+    common::IterableRange2D c_k = iterate_range2d(c_start, c_end);

```

Comment:
template argument is missing. Clang 10 output:
```
In file included from /Users/rasolca/code/DLA-Future-0/src/eigensolver/backtransformation/mc.cpp:1:
/Users/rasolca/code/DLA-Future-0/include/dlaf/eigensolver/backtransformation/mc.h:162:13: error: use of class template 'common::IterableRange2D' requires template arguments
    common::IterableRange2D c_k = iterate_range2d(c_start, c_end);
```

```suggestion
    auto c_k = iterate_range2d(c_start, c_end);
```